### PR TITLE
Release 1.0.0 — close the ways enforcement failed silently, and make setup two questions

### PR DESCRIPTION
**Pull request by chhhee10**
First stable release. `latest` has sat on 0.0.15 for the whole beta line, because `publish.yml` resolves the dist-tag as *next from a branch, beta for a prerelease on main, else latest* — so every `1.0.0-beta.N` went to `beta`. Dropping the suffix is what moves `latest`, which means `npm install -g failproofai` and the bare `npx -y failproofai` in every installed hook command start resolving here instead of to 0.0.15.

## The theme

Every fix below is one bug wearing different clothes: **the user's belief about their protection diverged from reality and nothing reconciled the two.** None of them were crashes or test failures — all are success paths, which is why none were caught.

## Enforcement that wasn't

- **PostToolUse denies were dropped on codex and copilot.** Both read a top-level `{decision:"block", reason}`; we sent `hookSpecificOutput.additionalContext`, which neither reads. So the policy ran, the verdict was logged, the dashboard counted it as enforcement — and nothing happened. Verified rather than inferred: a live A/B probe on codex 0.147.0 (block shape → `hook: PostToolUse Blocked`, the reason replaces the tool result and the model never sees the real output; our shape → `Completed`, model reads it verbatim), and a re-read of the shipped copilot 1.0.78 bundle where both `postToolUse` sites gate on `vK = t => t?.decision==="block" && typeof t.reason==="string"`.
- **`mode: "oss"` never stopped the daemon.** `--disconnect` writes the flag and claims "every cloud code path keys off this flag" — true of the CLI, false of the daemon, which holds the socket and had never read it. It now reads `mode.kind` (the shape `fp-config.ts` actually persists; reading it as a string is why an earlier cut of this fired zero times).
- **A revoked key reported as connected.** `--status` read the credential file, which records `--connect` time and is never revisited. The uploader already writes its verdict into the parked batch's filename and already declines to retry a definitive refusal — nothing read that directory. Now `--status` reports from it, and the same verdict fires once per session at `SessionStart`, so it surfaces with no flag and no user action.
- **A Hermes session arrived under two agent ids.** Confirmed on a customer org from a single collector. The id was derived from the session's own `cwd`/`source` and re-read every poll, and Hermes rewrites those throughout a run — so a session split the moment one changed. Identity now comes from which database the session is in, which cannot change mid-poll. Matches the standalone collector's scheme, so migrating machines are not renamed.

## Setup is two questions

`failproofai config` opens with **Recommended vs Customize** instead of asking scope, bundles, harnesses and cloud of somebody who just installed the tool. Recommended is a decision taken on their behalf — global scope, detected CLIs, a named 15-policy set — not a shortcut past the decisions. Customize is the previous wizard, unchanged.

`RECOMMENDED_POLICIES` is written out with its reasoning rather than derived from `defaultEnabled`, and includes three that were off and should not have been: `block-rm-rf`, `block-force-push`, `block-secrets-write`. It **unions** with what was already enabled, because `installHooks` runs with `replace: true` and writing the bare 15 would turn "give me the defaults" into a reduction in protection.

Setup's prose is down by two thirds. The consent clause stays — this is the only place transcripts are ever disclosed.

## Tests that could not fail

Two are worth calling out because they were worse than no test:

- The `mode` veto's fixtures carried the same wrong assumption as the code, so they reported a broken case as covered. Fixing them surfaced five more tests that read the *developer's own* `~/.failproofai/config.json` — passing on a machine that had not been set up, failing on one that had.
- Every Hermes test polled once. The bug only appears across two polls, which is why it shipped behind a green suite. The regression test polls twice with `cwd` rewritten in between.

## Gates

3456 TS tests · 28 Rust suites · tsc, clippy and fmt clean.

## Known, and deliberately not in this PR

- `enforcement-capability.ts`'s codex rows cite source paths that no longer exist in 0.147.0. Marked unverified rather than left looking like evidence — a re-probe is outstanding.
- The bundle vocabulary has a hole: four `defaultEnabled` policies are in no preset, so a first run that picks bundles omits the two that stop the agent disabling failproofai. Recommended routes around it; the bundles still have it.
- The hook source still derives its agent id from `cwd`, so it can disagree with the Hermes session source. Same instability class, not addressed here.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01HEwcerc9jE7ZBkfBiYRbep

## Hermes review

| Field | Value |
|---|---|
| Status | **Changes requested** |
| Reviewed commit | `3242b91f4d9cf5e9fae42c6d19736ed03f3488a1` |
| Policy revision | `1d8f31d926828f3bae215c58f5b35baa44acbff0` |
| Model | `gpt-5.6-terra` |
| Duration | 289s |
| Updated | 2026-08-12T11:33:05.412958812+00:00 |

### Summary

The PR fixes the prior default-Claude fallback, but Recommended setup can now finish with no hooks at all when no supported CLI is detected. Static parsing of modified TypeScript passed in an isolated container; dependency-backed tests could not run because the isolated container has no cached packages and network access is disabled.

### Changes

- Releases version 1.0.0 and updates Cloud branding.
- Adds recommended global setup and policy presets.
- Fixes Codex/Copilot PostToolUse deny payloads.
- Adds rejected-delivery health reporting.
- Makes daemon OSS mode and Hermes profile identities behave consistently.

### Validation

- `Passed` `docker run --rm --network=none -v /review/input/workspace:/workspace:ro -w /workspace oven/bun:latest sh -lc 'bun build src/hooks/configure-wizard.ts --target=node --external="*" --outfile=/tmp/configure-wizard.js && bun build src/hooks/delivery-health.ts --target=node --external="*" --outfile=/tmp/delivery-health.js && bun build src/hooks/policy-evaluator.ts --target=node --external="*" --outfile=/tmp/policy-evaluator.js'` — All three modified TypeScript modules parsed and bundled successfully in an isolated container. (0s)
- `Skipped` `docker run --rm --network=none -v /review/input/workspace:/workspace:ro -w /tmp oven/bun:latest sh -lc 'cp -a /workspace review && cd review && bun install --frozen-lockfile --offline && bunx vitest run __tests__/hooks/configure-wizard.test.ts __tests__/hooks/delivery-health.test.ts'` — The isolated container has no dependency cache; installing requires registry access, which is intentionally disabled by the harness network isolation. (0s)

### Findings

- **High/High** Recommended setup silently applies no protection when no CLI is detected — `recommendedClis` comes only from `detectInstalledClis()` (src/hooks/configure-wizard.ts:1122-1124). Recommended mode assigns that list directly to `clisSel` (line 1260), while the apply loop calls `installHooks` only when `clisForScope.length > 0` (line 1791). With no detected supported CLI, the wizard still completes and marks the daemon configured, but installs no hooks and never persists the recommended policy set. (`src/hooks/configure-wizard.ts:1260`)

### Open questions

None.

### Policy overrides

None.

## Summary by CodeRabbit

* **New Features**
  * Released FailproofAI 1.0.0.
  * Added Recommended and Customize setup modes with policy previews and preservation of existing policies.
  * Added delivery-health reporting, including rejected batches and credential-related issues.
  * Added session-start warnings for rejected cloud deliveries.
  * Improved Hermes profile identity stability.
  * Added fail-closed enforcement verification, unsupported-platform safeguards, and OSS cloud-disconnect enforcement.
* **Bug Fixes**
  * Corrected tool-blocking responses for Codex and Copilot integrations.
* **Documentation**
  * Updated cloud branding and Codex capability guidance.

### CODE REVIEW

**Comment by hermes-exosphere** [comment]
## Hermes

| | |
|---|---|
| Status | **Reviewing** |
| Verdict | Not reviewed yet |
| Head | `a7b5eaef16b6` |
| Rounds | 0 of 5 |

_No summary yet._

### What this changes

_No component map for this revision._

### Rounds

_No review has finished on this pull request yet._

### Findings

Nothing raised yet.

---

`@hermes-exosphere help` lists every command. This comment is maintained in place — I rewrite it after each review rather than posting a new one.

**Comment by hermes-exosphere** [comment]
## Hermes

| | |
|---|---|
| Status | **Reviewed** |
| Verdict | Changes requested |
| Head | `3242b91f4d9c` |
| Rounds | 1 of 5 |

The PR fixes the prior default-Claude fallback, but Recommended setup can now finish with no hooks at all when no supported CLI is detected. Static parsing of modified TypeScript passed in an isolated container; dependency-backed tests could not run because the isolated container has no cached packages and network access is disabled.

### What this changes

```mermaid
flowchart LR
    n0Setupwizard["~ Setup wizard"]
    n1Policypresets["~ Policy presets"]
    n2Hookenforcement["~ Hook enforcement"]
    n3Deliveryhealthreporting["+ Delivery health reporting"]
    n4Cloudpolicydaemon["~ Cloud policy daemon"]
    n5Hermescollector["~ Hermes collector"]
    n0Setupwizard -- "selects recommended policies" --> n1Policypresets
    n0Setupwizard -- "installs CLI hook configs" --> n2Hookenforcement
    n0Setupwizard -- "installs daemon service" --> n4Cloudpolicydaemon
    n3Deliveryhealthreporting -- "emits SessionStart warnings" --> n2Hookenforcement
    n4Cloudpolicydaemon -- "supplies managed policies" --> n2Hookenforcement
    n5Hermescollector -- "ships session batches" --> n3Deliveryhealthreporting
```

### Rounds

| Round | Reviewed | Commits in this round | Verdict |
|---|---|---|---|
| 0 | `a7b5eaef16b6` | `10edea80561c` `83949d908ffb` `79907c4cef51` `8a43d8b53306` `b3159f8ec491` `cb21e42f57ea` `a7b5eaef16b6` | Approved |
| 1 | `3242b91f4d9c` | `25eced7848e9` `3242b91f4d9c` | Changes requested — F1 |

### Findings

**Open**

- **F1** Recommended setup silently applies no protection when no CLI is detected (`src/hooks/configure-wizard.ts`) — round 1

---

`@hermes-exosphere help` lists every command. This comment is maintained in place — I rewrite it after each review rather than posting a new one.

**Review by hermes-exosphere — DISMISSED** [review]
Hermes found no blocking issues in this revision.

<details>
<summary>1 advisory finding</summary>

- **Medium/High** Recommended setup falls back to Claude when no CLI was detected — `recommendedClis` is derived solely from `detectInstalledClis()` and is assigned to `clisSel` at src/hooks/configure-wizard.ts:1260. If detection returns [], the resulting empty `clis` array is passed to `installHooks`; its existing default at src/hooks/manager.ts:167 interprets an empty array as `["claude"]`. Thus choosing Recommended on a machine with no detected CLI still creates/configures Claude hooks, contrary to the stated detected-CLIs-only behavior. (`src/hooks/configure-wizard.ts:1260`)

</details>

File: src/hooks/configure-wizard.ts [hunk 1]
```diff
@@ -1158,6 +1250,15 @@ export async function runConfigureWizard(io: WizardIO = {}): Promise<WizardResul
...
    */
   const carried: { clis: string[] | null } = { clis: null };
+  // The recommended path answers both questions here, which is what SKIPS the
+  // loop below — its condition is `clisSel === null` and this fills it in.
+  // Written as a pre-fill rather than as an extra clause on the loop condition
+  // so the invariant the rest of the function depends on ("past this loop, both
+  // are assigned") stays provable by the compiler rather than by argument.
+  if (mode === "recommended") {
+    presets = [];
+    clisSel = recommendedClis;
```
Comment (hermes-exosphere) [inline src/hooks/configure-wizard.ts:1260]: **Hermes — High/High (`COR-001`): Recommended setup silently applies no protection when no CLI is detected**

`recommendedClis` comes only from `detectInstalledClis()` (src/hooks/configure-wizard.ts:1122-1124). Recommended mode assigns that list directly to `clisSel` (line 1260), while the apply loop calls `installHooks` only when `clisForScope.length > 0` (line 1791). With no detected supported CLI, the wizard still completes and marks the daemon configured, but installs no hooks and never persists the recommended policy set.

Required change: Do not permit a successful Recommended apply with an empty detected-CLI list: direct the user to Customize/select a supported CLI, or require a detected CLI before continuing. Add a regression test for zero detected CLIs.

**Review by hermes-exosphere — DISMISSED** [review]
Hermes found blocking issues that should be addressed.

### High: Recommended setup silently applies no protection when no CLI is detected

- Rule: `COR-001`
- Location: `src/hooks/configure-wizard.ts:1260`
- Evidence: `recommendedClis` comes only from `detectInstalledClis()` (src/hooks/configure-wizard.ts:1122-1124). Recommended mode assigns that list directly to `clisSel` (line 1260), while the apply loop calls `installHooks` only when `clisForScope.length > 0` (line 1791). With no detected supported CLI, the wizard still completes and marks the daemon configured, but installs no hooks and never persists the recommended policy set.
- Required change: Do not permit a successful Recommended apply with an empty detected-CLI list: direct the user to Customize/select a supported CLI, or require a detected CLI before continuing. Add a regression test for zero detected CLIs.

**Review by NiveditJain — APPROVED** [review]
